### DISCUSSION TITLE
A little help please

### DISCUSSION THREAD

**Discussion by RubiXKube** [calibx/msfsdeck#114] [Q&A] [answered]
Hi,
I'm using the Honda Jet from FlightFX in MSFS,
There is a little bug with a loopedeck encoder only with this plane
I can't set an autopilot altitude with the encoder, it's always go back to previous value

I try to find a solution but i'm not developer
Honda jet developper ask me...
------------
Marwan Gharib — Aujourd’hui à 17:54(on flight fx discord)
Hello, what default sim event is the loopdeck live triggering?
specifically these are the commands used for the HJET
ALT SEL KNOB -> INCREASE/DECREASE AUTOPILOT REFERENCE ALTITUDE
other commands might not work as intended as they don't change the needed variables
------------

I don't have any answer to this, i don't know how your plugins works..
Could you help me ?

Thanks !

**Comment by calibx** [comment, marked as answer]
Hi,
I use the AP_ALT_VAR_SET_ENGLISH event from SimConnect.
I don't see the HJET ALT SEL KNOB event in the MSFS Wiki : https://docs.flightsimulator.com/html/Programming_Tools/SimVars/Simulation_Variables.htm
Maybe it needs a WASM module to be set.
Unfortunatly, this plugin doesn't come with a WASM module as it is way more complicated to develop and deliver with the plugin system.

A solution for you would be to use another wrapper between the loupedeck and MSFS for that control like FSUIPC, Mobiflight or Spadnext. You will loose feedback values but you will be able to trigger what you want as they come with a WASM module.

**Reply by RubiXKube** [reply]
Thank you for your answer,
HJET ALT SEL KNOB is L:VAR from this jet

I thought to try with FSUIPC, but HJET dev was online when i gave him your answer
He found a solution in 1 minutes
Hjet will be fully working with your plugin now !!

Thank you so much !

**Reply by calibx** [reply]
Good news ! Happy for U !

**Reply by belleair** [reply]
RubiXKube.....im having the same issue with the Hjet whats the solution?

**Reply by RubiXKube** [reply]
> RubiXKube.....im having the same issue with the Hjet whats the solution?

You have to find HPFD.js in Hjet folder(ffx-hjet), it depends your game version and where you bought Hjet
for me it's
C:\Users\RubiX\AppData\Roaming\Microsoft Flight Simulator\Packages\Official\Steam\ffx-hjet\html_ui\Pages\VCockpit\Instruments\HA420_Systems\HPFD

Then in HPFD.js, 2 things to do

find this (line about 3940)
            //hjet custom
            this.interceptKey('AP_ALT_VAR_INC', false)
            this.interceptKey('AP_ALT_VAR_DEC', false)

Add below
             this.interceptKey('AP_ALT_VAR_SET_ENGLISH', false) //add this

And find this (line about 4090)
            /////////// HJET
            case 'AP_ALT_VAR_INC':
                this.increaseHJETALT(window.framRateCounter)
                break;
            case 'AP_ALT_VAR_DEC':
                this.decreaseHJETALT(window.framRateCounter)
                break;

Add below
            case 'AP_ALT_VAR_SET_ENGLISH'://add this 
                if (value !== undefined) {//add this 
                    SimVar.SetSimVarValue("L:HJET_AP_ALT_VAR", "number", value) //add this 
                }//add this 
                break;//add this 

It will be included in next update
hope it will help you

**Reply by belleair** [reply]
thanks for the tip.....i was able to find the lines stated above but i dont have a freaking clue what im doing can you send screenshots of what you did to fix the issue? thanks in advance

**Reply by RubiXKube** [reply]
![Screenshot_1](https://user-images.githubusercontent.com/66200975/205977288-00d09749-ef05-47da-b972-1df1c4595696.png)
![Screenshot_2](https://user-images.githubusercontent.com/66200975/205977316-12f429c4-1625-48bd-a4d1-d4c37a1222b6.png)

**Reply by belleair** [reply]
it worked thank you so much!!!

**Comment by RubiXKube** [comment]
Just a little thing now,
when i use encode on HJET when you go above 0
it goes to FL1000, and can't come back to 0 in other way

may be add negative value on this encoder ?

**Comment by calibx** [comment]
Hi, 
Which control are you using ?
My ap alt control can go from -10000 to 99900.
Can you test it on another plane to confirm ?

**Comment by RubiXKube** [comment]
You're right, in other plane your encoder go in negative value
And plane stop to zero. sorry :)

**Reply by calibx** [reply]
Maybe HJET interpret the alt values changes to convert them in FL.
I can't find any event relating to the FL.
Maybe the dev of HJEt may help

**Reply by belleair** [reply]
it worked!!! you're a saint!!!
